koku/masu/util/aws/common.py

```diff
@@ -758,6 +768,33 @@ def delete_s3_objects(request_id, keys_to_delete, context) -> list[str]:
     return []
 
 
+def clear_s3_files(csv_s3_path, provider_uuid, start_date, context, request_id):
+    """Clear s3 files for daily archive processing AWS/Azure ONLY"""
+    account = context.get("account")
+    provider_type = context.get("provider_type")
+    parquet_path_s3 = get_path_prefix(account, provider_type, provider_uuid, start_date, "parquet")
+    parquet_daily_path_s3 = get_path_prefix(
+        account, provider_type, provider_uuid, start_date, "parquet", report_type="raw", daily=True
+    )
+    parquet_ocp_on_cloud_path_s3 = get_path_prefix(
+        account, provider_type, provider_uuid, start_date, "parquet", report_type="openshift", daily=True
+    )
+    s3_prefixes = []
+    dh = DateHelper()
+    list_days = dh.list_days(start_date, dh.now.replace(tzinfo=None))
+    for day in list_days:
+        path = f"/{day.date()}"
+        s3_prefixes.append(csv_s3_path + path)
+        s3_prefixes.append(parquet_path_s3 + path)
+        s3_prefixes.append(parquet_daily_path_s3 + path)
+        s3_prefixes.append(parquet_ocp_on_cloud_path_s3 + path)
+    to_delete = []
+    for prefix in s3_prefixes:
+        for obj_summary in _get_s3_objects(prefix):
+            to_delete.append(obj_summary.Object().key)
```

Comment:
This could be written differently to avoid the intermediate list building and mutating as well as string concatenation. Not a blocker, just a different style example.
```suggestion
    paths = (
        csv_s3_path,
        parquet_path_s3,
        parquet_daily_path_s3,
        parquet_ocp_on_cloud_path_s3,
    )
    to_delete = []
    dh = DateHelper()
    list_days = dh.list_days(start_date, dh.now.replace(tzinfo=None))
    for day in list_days:
        day = day.date()
        to_delete.extend(
            obj_summary.Object().key
            for path in paths
            for obj_summary in _get_s3_objects(f"{path}/{day}")
        )
```